.github/workflows/main.yml

```diff
@@ -7,7 +7,7 @@ on:
 
 jobs:
   run-ci:
-    runs-on: windows-latest
+    runs-on: macos-latest
```

Comment:
Can we run tests on both?

Reply:
When I run tests on both windows and macos, the runners are in a queue for longer than a day and the github actions don't end up running.
![image](https://github.com/OpenAdaptAI/OpenAdapt/assets/43456930/048317d5-cdb5-40b4-bd97-b374d544b5c3)

Reply:
Any idea why this happens? What does the log output look like? Can you please paste a link? Does this also happen if we do Windows and MacOS separately?

Reply:
This is the link to the [screenshot](https://github.com/Mustaballer/puterbot/actions/runs/5417204943). I found a new structure using [matrix strategy](https://adamtheautomator.com/github-actions-matrix/) and hopefully that might resolve the issues.